libnetwork/resolver.go

```diff
@@ -351,29 +354,36 @@ func (r *Resolver) serveDNS(w dns.ResponseWriter, query *dns.Msg) {
 	queryName := query.Question[0].Name
 	queryType := query.Question[0].Qtype
 
+	ctx, span := otel.Tracer("").Start(context.Background(), "resolver.serveDNS", trace.WithAttributes(
```

Comment:
`context.TODO()`

Reply:
Hm I don't think that make sense. IMO `context.TODO()` marks place where we might end up with a parent context, if we do all the plumbing required. But that's not the case here: `serveDNS` is the entrypoint for any DNS request received by the embedded resolver. In that regard, it's very much like what the http package does: it initializes a new Context for every request. And this specific span is actually a root span.